core/dictgen/src/rootcling_impl.cxx

```diff
@@ -2003,8 +2003,6 @@ static bool WriteAST(StringRef fileName, clang::CompilerInstance *compilerInstan
 
    // Make sure it hits disk now.
    out->flush();
-   bool deleteOutputFile = compilerInstance->getDiagnostics().hasErrorOccurred();
-   compilerInstance->clearOutputFiles(deleteOutputFile);
```

Comment:
Did you verify (i.e. try) that the PCH file gets removed?

Reply:
The problem is the inverse. Clang creates a temporary file and then makes moves it to the requested filename. So, yes, the generated file is with the expected name. Otherwise we hit an assert.

Reply:
I don't understand your reply.

In case of failure we don't want to have output files left after the process ends, even if these files were temporary. `clearOutputFiles` was taking care of that.

Did you verify (i.e. try) that the PCH file gets removed *in the error case*?

Reply:
This piece of code did not disappear, it is still present in the dtor of cling::Interpreter: https://github.com/root-project/root/pull/4675/files#diff-8355e949fdbd024ed856aa110ff93ec9R608

So that's a NFC, essentially.